pkg/probe/turbo_probe.go

```diff
@@ -70,38 +73,32 @@ func newTurboProbe(probeConf *ProbeConfig) (*TurboProbe, error) {
 
 	myProbe := &TurboProbe{
 		ProbeConfiguration: probeConf,
-		DiscoveryClientMap: make(map[string]*TargetDiscoveryAgent),
+		DiscoveryClient:    nil, // DiscoveryClient is originally empty. Should be initialized later.
 		RegistrationClient: NewProbeRegistrator(),
+		TargetsToAdd:       map[string]bool{},
 	}
 
 	glog.V(2).Infof("[NewTurboProbe] Created TurboProbe: %v", myProbe)
 	return myProbe, nil
 }
 
-func (theProbe *TurboProbe) getDiscoveryClient(targetIdentifier string) TurboDiscoveryClient {
-	target, exists := theProbe.DiscoveryClientMap[targetIdentifier]
-
-	if !exists {
-		glog.Errorf("[GetTurboDiscoveryClient] Cannot find Target for address: %s", targetIdentifier)
+func (theProbe *TurboProbe) getDiscoveryClient() TurboDiscoveryClient {
+	if theProbe.DiscoveryClient == nil {
+		glog.Errorf("[GetTurboDiscoveryClient] No DiscoveryAgent configured.")
 		return nil
 	}
-	return target.TurboDiscoveryClient
+
+	return theProbe.DiscoveryClient.TurboDiscoveryClient
 }
 
 // TODO: this method should be synchronized
-func (theProbe *TurboProbe) GetTurboDiscoveryClient(accountValues []*proto.AccountValue) TurboDiscoveryClient {
-	var address string
-	identifyingField := theProbe.RegistrationClient.GetIdentifyingFields()
-
-	address = findTargetId(accountValues, identifyingField)
-	target, exists := theProbe.DiscoveryClientMap[address]
-	if !exists {
-		glog.Errorf("[GetTurboDiscoveryClient] Cannot find Target for address: %s", address)
-		//TODO: CreateDiscoveryClient(address, accountValues, )
+func (theProbe *TurboProbe) GetTurboDiscoveryClient() TurboDiscoveryClient {
```

Comment:
This function seems to be exactly the same as the above one? The only difference is this one is exported to be accessed from other package. Maybe we should remove the above one?